### ISSUE TITLE
When katago is playing, the result shows that katago is illegal

### ISSUE THREAD

**Issue by dergo853** [lightvector/KataGo#681]
When Katago was almost finished a game, it sometimes shows that katago was illegal.
For example, katago is black, and the result is W+illegal.
There is a sgf file: http://www.yss-aya.com/cgos/viewer.cgi?9x9/SGF/2022/09/09/1345415.sgf
http://www.yss-aya.com/cgos/9x9/SGF/2022/09/09/1345415.sgf
How can I solve it?

**Comment by lightvector** [comment]
You probably configured KataGo to use the wrong rules. Check CGOS's documentation for what its rules are, then specify those rules in KataGo's config.

**Comment by dergo853** [comment]
CGOS's rule is tromp-taylor, and config's rule is tt, too.

**Comment by lightvector** [comment]
No, I remember from before CGOS's rule is not tromp-taylor, it's something else. Please check the documentation for CGOS carefully.

**Comment by dergo853** [comment]
aha? I will check again carefully.

**Comment by dergo853** [comment]
Programs that play on CGOS use rules very similar to the [Tromp/Taylor](http://senseis.xmp.net/?LogicalRules) rules (sometimes referred to as "The Logical Rules") but prohibit suicide. These are also rather similar to Chinese rules but the games are played out to the end, then no further stones are removed, and the ko rules are clarified to be Positional Superko.
So, should I use Chinese or tt rules?

**Comment by lightvector** [comment]
Seems like you should use neither one. Since CGOS uses neither Chinese rules nor Tromp-Taylor rules, but instead uses an alternative version of Tromp Taylor rules where suicide is prohibited, you need to also manually specify every separate aspect of the rules rule (scoring, ko, suicide, etc) to the way it should be to make it match.

**Comment by dergo853** [comment]
Oh. How should I do?

**Comment by lightvector** [comment]
Take a look at this section in the config.
https://github.com/lightvector/KataGo/blob/master/cpp/configs/gtp_example.cfg#L94-L115

**Comment by hwj-111** [comment]
in your KATAGO config file, locate "multiStoneSuicideLegal = true" and change it to "multiStoneSuicideLegal = false".  Then you should be fine when playing on CGOS

**Comment by dergo853** [comment]
Thanks a lot. @lightvector @hwj-111

**Comment by dergo853** [comment]
but, it shows "Cannot both specify 'rules' and individual rules like multiStoneSuicideLegal"

**Comment by dergo853** [comment]
??

**Comment by OmnipotentEntity** [comment]
This was explained a little bit up thread:

> Since CGOS uses neither Chinese rules nor Tromp-Taylor rules, but instead uses an alternative version of Tromp Taylor rules where suicide is prohibited, you need to also manually specify every separate aspect of the rules rule (scoring, ko, suicide, etc) to the way it should be to make it match.

For CGOS, if it's just Tromp Taylor without suicide you'll probably want:

```
koRule = POSITIONAL   # Positional superko
scoringRule = TERRITORY  # Territory scoring (uses a sort of special computer-friendly territory ruleset)
taxRule = NONE  # All surrounded empty points are scored
multiStoneSuicideLegal = false  # Is multiple-stone suicide legal? (Single-stone suicide is always illegal).
hasButton = false # Set to true when area scoring to award 0.5 points to the first pass.
friendlyPassOk = false  # Set to true except for computer rulesets that requires capturing all stones before passing.
whiteHandicapBonus = 0    # In handicap games, give white no compensation for black's handicap stones (Tromp-taylor, NZ, JP)
```

See https://lightvector.github.io/KataGo/rules.html for details on what TT rules actually are.  I just took these rules and changed the suicide rule from true to false.

Disclaimer, I have never run a bot on CGOS.

**Comment by lightvector** [comment]
@OmnipotentEntity Thanks! I think `scoringRule` should be `AREA` instead since if you're tromp-taylor rules with a modification, then you're going to be area scoring as a baseline? @dergo853 - sorry for the long back and forth, see if that gets things working for you?

**Comment by OmnipotentEntity** [comment]
Good catch. Fixed. I knew that but somehow deleted the wrong line on accident.

**Comment by dergo853** [comment]
ok. i'll try
